js/store.js

```diff
@@ -292,29 +291,23 @@ const store = {
       thirdParty['firstPartyHostnames'].push(origin);
     }
 
-    // add link in first party if it doesn't exist yet
-    // and the third party is visible (i.e. not allowlisted)
-    if (!this.isFirstPartyLinkedToThirdParty(firstParty, target)) {
```

Comment:
`isFirstPartyLinkedToThirdParty` has been removed on the ` if (!thirdParty['isVisible'] && !this.onAllowList(origin, target)) ` section of your code, won't this make updates overfire?